src/actions/MapActions.js

```diff
@@ -28,18 +36,34 @@ export const initializeMap = ({region}) => {
 
         const unsubscribeApprovedCampsitesSnapshot = firebase.firestore().collection('campsites')
             .onSnapshot(querySnapshot => {
-                const sites = _.map(querySnapshot.docs, (doc, index) => {
-                    let preparedSite = _.clone(doc.data());
+                let sites = [];
+                let imageDataArray = [];
+
+                _.forEach(querySnapshot.docs, (doc) => {
+                    const docData = doc.data();
+
+                    const imageData = [doc.id, docData.siteImageData];
+                    imageDataArray.push(imageData);
+
+                    let preparedSite = _.clone(docData);
+                    delete preparedSite.siteImageData;
                     preparedSite.id = doc.id;
                     preparedSite.key = preparedSite.id;
 
-                    return preparedSite;
+                    sites.push(preparedSite);
```

Comment:
Now when we get the site data from Firebase, we pass along `sites` as an array of all the sites without their `siteImageData`. Separately we create an array of key/values with the site `id` and the `siteImageData` and store it all to `AsyncStorage` for local access.